osu.Game/Overlays/BeatmapSet/BeatmapRulesetTabItem.cs

```diff
@@ -3,143 +3,87 @@
-                                    Anchor = Anchor.Centre,
-                                    Origin = Anchor.Centre,
-                                    Margin = new MarginPadding { Horizontal = 5f },
-                                    Font = OsuFont.Default.With(weight: FontWeight.SemiBold),
-                                }
-                            }
-                        }
+                        Anchor = Anchor.Centre,
+                        Origin = Anchor.Centre,
+                        Margin = new MarginPadding { Horizontal = 5f },
+                        Font = OsuFont.Default.With(weight: FontWeight.SemiBold),
+                        Colour = colourProvider.Foreground1,
                     }
-                },
-                bar = new Box
-                {
-                    Anchor = Anchor.BottomCentre,
-                    Origin = Anchor.BottomCentre,
-                    RelativeSizeAxes = Axes.X,
-                },
-                new HoverClickSounds(),
-            };
+                }
+            });
+        }
+
+        protected override void LoadComplete()
+        {
+            base.LoadComplete();
 
             BeatmapSet.BindValueChanged(setInfo =>
             {
                 var beatmapsCount = setInfo.NewValue?.Beatmaps.Count(b => b.Ruleset.Equals(Value)) ?? 0;
 
                 count.Text = beatmapsCount.ToString();
-                count.Alpha = beatmapsCount > 0 ? 1f : 0f;
+                countContainer.FadeTo(beatmapsCount > 0 ? 1 : 0);
 
                 Enabled.Value = beatmapsCount > 0;
             }, true);
 
-            Enabled.BindValueChanged(v => nameContainer.Alpha = v.NewValue ? 1f : 0.5f, true);
-        }
-
-        [Resolved]
-        private OsuColour colour { get; set; }
-
-        protected override void LoadComplete()
-        {
-            base.LoadComplete();
-
-            count.Colour = colour.Gray9;
-            bar.Colour = colour.Blue;
-
-            updateState();
-        }
-
-        private void updateState()
-        {
-            var isHoveredOrActive = IsHovered || Active.Value;
-
-            bar.ResizeHeightTo(isHoveredOrActive ? 4 : 0, 200, Easing.OutQuint);
-
-            name.Colour = isHoveredOrActive ? colour.GrayE : colour.GrayC;
-            name.Font = name.Font.With(weight: Active.Value ? FontWeight.Bold : FontWeight.Regular);
-        }
-
-        #region Hovering and activation logic
-
-        protected override void OnActivated() => updateState();
-
-        protected override void OnDeactivated() => updateState();
+            Enabled.BindValueChanged(enabled =>
+            {
+                if (enabled.NewValue)
+                {
+                    UpdateState();
+                    return;
+                }
 
-        protected override bool OnHover(HoverEvent e)
-        {
-            updateState();
-            return false;
+                AccentColour = colourProvider.Foreground1;
+            }, true);
```

Comment:
I'm also really confused by this part too. I can't tell what this is trying to achieve or why `AccentColour` is set here, and I don't think I should be digging in this code to find out.